rumqttc/src/state.rs

```diff
@@ -193,11 +210,14 @@ impl MqttState {
             error!("Unsolicited suback packet: {:?}", suback.pkid);
             return Err(StateError::Unsolicited(suback.pkid));
         }
-
-        let tx = self
+        // No expecting ordered acks for suback
+        // Search outgoing_sub to find the right suback.pkid
+        let pos = self
             .outgoing_sub
-            .remove(&suback.pkid)
+            .iter()
+            .position(|(pkid, _)| *pkid == suback.pkid)
             .ok_or(StateError::Unsolicited(suback.pkid))?;
+        let (_, tx) = self.outgoing_sub.remove(pos).unwrap();
```

Comment:
```suggestion
        let (_, tx) = self.outgoing_sub.swap_remove_back(pos).unwrap();
```

Since the ordering is not important, could we maybe just swap_remove the element?